ksql-serde/src/test/java/io/confluent/ksql/serde/json/KsqlJsonDeserializerTest.java

```diff
@@ -145,12 +165,31 @@ public void shouldCreateJsonStringForStructIfDefinedAsVarchar() throws JsonProce
     final Schema schema = SchemaBuilder.struct()
         .field("itemid".toUpperCase(), Schema.OPTIONAL_STRING_SCHEMA)
         .build();
-    final KsqlJsonDeserializer deserializer = new KsqlJsonDeserializer(schema, false);
+    final KsqlJsonDeserializer deserializer = new KsqlJsonDeserializer(
+        schema,
+        false,
+        recordLogger);
 
     final GenericRow expected = new GenericRow(Collections.singletonList(
         "{\"CATEGORY\":{\"ID\":2,\"NAME\":\"Food\"},\"ITEMID\":6,\"NAME\":\"Item_6\"}"));
     final GenericRow genericRow = deserializer.deserialize("", "{\"itemid\":{\"CATEGORY\":{\"ID\":2,\"NAME\":\"Food\"},\"ITEMID\":6,\"NAME\":\"Item_6\"}}".getBytes(StandardCharsets.UTF_8));
     assertThat(genericRow, equalTo(expected));
   }
 
+  @Test
+  public void shouldLogDeserializationErrors() {
+    // When:
+    Throwable cause = null;
+    final byte[] data = "{foo".getBytes(StandardCharsets.UTF_8);
+    try {
+      ksqlJsonDeserializer.deserialize("", data);
```

Comment:
Perhaps slip in a `fail()` under this line (as you have in the other tests) to make it more explicit that the deserialization should error?